packages/ws/src/utils/IdentifyThrottler.ts

```diff
@@ -10,20 +13,26 @@ export class IdentifyThrottler {
 	public constructor(private readonly manager: WebSocketManager) {}
 
 	public async waitForIdentify(): Promise<void> {
-		if (this.identifyState.remaining <= 0) {
-			const diff = this.identifyState.resetsAt - Date.now();
-			if (diff <= 5_000) {
-				const time = diff + Math.random() * 1_500;
-				await sleep(time);
+		await this.queue.wait();
+
+		try {
+			if (this.identifyState.remaining <= 0) {
+				const diff = this.identifyState.resetsAt - Date.now();
+				if (diff <= 5_000) {
+					const time = diff + Math.random() * 1_500;
```

Comment:
I know this comes from old code but... can you please explain why this *<sub>incredibly random</sub>* `Math.random()` is used here?

Reply:
Accounts for latency. `waitForIdentify` will always be called before we send an identify payload, which reaches Discord ever so slightly later.

Reply:
Can you write a comment above mentioning this, please? I feel like I'll forget in a couple of days and ask again whenever that line is touched 😂